File: app/src/main/java/com/tianchuang/ihome_b/base/BaseHolder.java
package com.tianchuang.ihome_b.base;

import android.view.View;

/**
 * Created by Abyss on 2016/11/15.
 */
public abstract class BaseHolder<T> {
    public View holderView;
    public BaseHolder() {
        //初始化holderview
        holderView = initHolderView();
        holderView.setTag(this);

    }
    /**
     * 获取holderview
     */
    public View getholderView() {
        return holderView;
    }
    public abstract View initHolderView();
    public abstract void bindData(T data);
}

File: app/src/main/java/com/tianchuang/ihome_b/base/BaseHolder.kt
package com.tianchuang.ihome_b.base

import android.view.View

/**
 * Created by Abyss on 2016/11/15.
 */
abstract class BaseHolder<in T> {
    var holderView: View

    init {
        //初始化holderview
        holderView = initHolderView()
        holderView.tag = this
    }

    /**
     * 获取holderview
     */
    fun getholderView(): View {
        return holderView
    }

    abstract fun initHolderView(): View
    abstract fun bindData(data: T)
}

